### PR TITLE
[hygiene] Volta pin latest node / yarn

### PR DESCRIPTION
**Pull request by stefanpenner**

### CODE REVIEW

**Comment by stefanpenner** [comment]
@ef4 any concerns with this? Or do we keep the version of node / yarn more conservative for a reason?

File: package.json [hunk 1]
```diff
@@ -84,8 +84,8 @@
     "npm": false
   },
   "volta": {
-    "node": "12.16.1",
-    "yarn": "1.22.5"
+    "node": "16.3.0",
```
[the remark this answers is not in the document]
Comment (stefanpenner) [reply package.json:87]: @rwjblue let's sync up.

**Comment by rwjblue** [comment]
We'll also have to fix up our `exports` or the imports that use things that aren't included in `exports`, on Node 16 the `exports` property is enforced. CI is failing for this reason:

```
Package subpath './src/load-ember-template-compiler' is not defined by "exports" in /home/runner/work/embroider/embroider/node_modules/@embroider/core/package.json
```

**Comment by rwjblue** [comment]
https://github.com/embroider-build/embroider/pull/851 should address that last item

**Comment by rwjblue** [comment]
> Or do we keep the version of node / yarn more conservative for a reason?

@stefanpenner - We generally want to test on the "lowest supported platform" (to make sure we still support whatever is our oldest node), but as you can tell we have issues with **newer** Nodes too. Maybe we should have at least one set of tests that run with oldest and newest?

**Comment by ef4** [comment]
The test upgrade work that @thoov has started can address this. In ember-auto-import where we have the same concerns, we can switch node versions per scenario. https://github.com/ef4/ember-auto-import/blob/27485d4dc76de152b7bfb1efb49e5a06fc565197/test-scenarios/scenarios.ts#L24

**Comment by stefanpenner** [comment]
@ef4 I'll coordinate with @thoov

**Comment by stefanpenner** [comment]
Online conversations, everyone seemed ok with this choice as a baseline. We can always add additional test scenarios + node permutations.

**Pull request merged**